hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -487,9 +492,17 @@ protected void appendDataAndDeleteBlocks(Map<HeaderMetadataType, String> header,
             getUpdatedHeader(header, config), keyField));
       }
 
+      if (shouldWriteRecordPositions
+          && !recordsToDeleteWithPositions.isEmpty()
+          && StringUtils.isNullOrEmpty(baseFileInstantTimeForDeletePositions)) {
+        LOG.warn("Base file instant time of delete positions cannot be determined, possibly "
+            + "due to inserts or different base file instant times for the input records.");
+      }
+
       if (appendDeleteBlocks && !recordsToDeleteWithPositions.isEmpty()) {
-        blocks.add(new HoodieDeleteBlock(recordsToDeleteWithPositions, shouldWriteRecordPositions,
-            getUpdatedHeader(header, config)));
+        blocks.add(new HoodieDeleteBlock(
+            recordsToDeleteWithPositions, baseFileInstantTimeForDeletePositions,
```

Comment:
can you help me locate where we do add the base instant time for data block headers?